Swift/BRCrypto/BRCryptoSystem.swift

```diff
@@ -393,7 +393,7 @@ public final class System {
         self.listener  = listener
         self.account   = account
         self.onMainnet = onMainnet
-        self.path  = path
+        self.path  = path + (path.last == "/" ? "" : "/") + account.fileSystemIdentifier
```

Comment:
This actually led to an interesting quirk. In the Java demo app (I think the Swift too), when we re-create the System on a `reset`, we both read the path from the current system and use that for the new system.

Well, since we modify path, here, that means the path keeps growing ala:
* create initial system with path as "/test/"
* system sets path to "/test/<hash>"
* reset system, using the current system.path to pass to new system (i.e. "/test/<hash>")
* system sets path to "/test/<hash>/<hash>"
* reset again, using the current system.path to pass to new system (i.e. "/test/<hash>/<hash>")
* etc.

So, do we care? If so, do we want to leave path (as returned to the client) as is, and internally have theActualPath, which we don't return to the client?

I'll push the Java as is, then we can change (or not) both the Swift/Java depending on this discussion.

Reply:
Pushed a 'fix'.  The App needs to pass in the same stuff on `reset`.  So I saved the stuff {account, storagePath, query, ...} in `CoreDemoAppDelegate`